File: Lab_10/BankApp/Commands/WithdrawCommand.cs
using BankApp.App;

namespace BankApp.Commands;

public class WithdrawCommand : ICommand
{
    private Bank _bank;
    private string _accountNumber;
    private decimal _amount;

    public WithdrawCommand(Bank bank, string accountNumber, decimal amount)
    {
        _bank = bank;
        _accountNumber = accountNumber;
        _amount = amount;
    }

    public void Execute()
    {
        _bank.Withdraw(_accountNumber, _amount);
    }
    
    public void Signature()
    {
        Console.WriteLine($"Withdrawn {_amount} from Account No. {_accountNumber}");
    }
}

File: lab10/Bank Account Command Pattern/BankApp/src/Command/WithdrawCommand.java
package Command;

import App.bank;

public class WithdrawCommand implements Command {

    private App.bank bank;
    private String accountNumber;
    private double amount;
    public WithdrawCommand(bank bank, String accountNumber, double amount) {
        this.bank = bank;
        this.accountNumber = accountNumber;
        this.amount = amount;
    }


    @Override
    public void execute() {
        bank.deposit(accountNumber, amount);
    }

    public void signature() {
        System.out.println("Withdrawn " + amount + " from Account No. " + accountNumber);
    }


}
